src/modules/recipes/NewRecipeForm/index.jsx

```diff
@@ -0,0 +1,215 @@
+    const newRecipe = {
+      idMeal: Date.now().toString(),
+      strMeal: data.strMeal,
+      strCategory: data.strCategory,
+      strArea: data.strArea,
+      strInstructions: data.strInstructions,
+      strMealThumb: data.strMealImage,
+    };
+
+    // Malzemeleri ve ölçüleri ayırma işlemi
+    data.ingredients.forEach((item, index) => {
+      if (index < 20) {
+        // API'nin 20 malzeme ve ölçü sınırına uymak için
+        // strIngredient1, strIngredient2 vb. için malzemeleri ekler
+        newRecipe[`strIngredient${index + 1}`] = item.ingredient;
+
+        // strMeasure1, strMeasure2 vb. için ölçüleri ekler
+        newRecipe[`strMeasure${index + 1}`] = item.amount
+          ? `${item.amount} ${item.measure}`
+          : "";
+      }
+    });
+
+    // Eğer 20'den az malzeme varsa, kalan alanları boş yapar
+    for (let i = data.ingredients.length; i < 20; i++) {
+      newRecipe[`strIngredient${i + 1}`] = "";
+      newRecipe[`strMeasure${i + 1}`] = "";
+    }
+
+    dispatch({ type: RECIPE_ACTIONS.update, payload: [newRecipe] });
+    console.log("Form submitted:", newRecipe);
+  };
+
+  return (
+    <div className="form-container">
+      <h2>Add New Recipe</h2>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <div className="form-group">
+          <label htmlFor="strMeal">Meal Name</label>
+          <input
+            id="strMeal"
+            {...register("strMeal", {
+              minLength: {
+                value: 3,
+                message: "Meal name should be at least 3 characters.",
+              },
+              required: "Meal name is required.",
+            })}
+          />
+          {errors.strMeal && (
+            <p className="error-message">{errors.strMeal.message}</p>
+          )}
+        </div>
+
+        <div className="form-group">
+          <label htmlFor="strCategory">Select Category</label>
+          <select id="strCategory" {...register("strCategory")}>
+            {categories.map((category) => (
+              <option key={category} value={category}>
+                {category}
+              </option>
+            ))}
+          </select>
+        </div>
+
+        <div className="form-group">
+          <label htmlFor="strArea">Select Area</label>
+          <select id="strArea" {...register("strArea")}>
+            {areas.map((area) => (
+              <option key={area} value={area}>
+                {area}
+              </option>
+            ))}
+          </select>
+        </div>
+
+        <div className="form-group">
+          <label>Ingredients</label>
+          {fields.map((field, index) => (
+            <div key={field.id} className="ingredient-item">
+              <input
+                {...register(`ingredients.${index}.ingredient`, {
+                  required: "Ingredient is required.",
+                })}
+                placeholder="Ingredient"
+              />
+              <input
+                type="number"
```

Comment:
Use type text here instead so it supports values like 0.5 and 0,5. Make sure to add a validation pattern (ask Chatgpt or similar to generate it).